server/src/utils/getOrInitialiseSolFileEntry.ts

```diff
@@ -10,19 +12,13 @@ import { SolFileEntry } from "../parser/analyzer/SolFileEntry";
  * Uri needs to be decoded and without the "file://" prefix.
  */
 export function getOrInitialiseSolFileEntry(
-  {
-    projects,
-    solFileIndex,
-  }: {
-    projects: SolProjectMap;
-    solFileIndex: SolFileIndexMap;
-  },
+  serverState: ServerState,
   uri: string
 ): ISolFileEntry {
-  let solFileEntry = solFileIndex[uri];
+  let solFileEntry = serverState.solFileIndex[uri];
 
   if (solFileEntry === undefined) {
-    const project = findProjectFor({ projects }, uri);
+    const project = new ProjectlessProject(serverState, path.dirname(uri));
```

Comment:
Might want to check that use of `path`.

Reply:
I double checked but couldn't find any problem with it. The attribute `basePath` is stored in the os-specific path format and operations with it are performed using the `path` module. Should I test a specific scenario where it could fall short ?

Reply:
Cool. If we are happy our integration tests cover this across OSes, no change needed.